service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -580,4 +592,73 @@ public void deleteSiteImage(JWTData userData, int imageId) {
 
     db.deleteFrom(SITE_IMAGES).where(SITE_IMAGES.ID.eq(imageId)).execute();
   }
+
+  @Override
+  public List<FilterSitesResponse> filterSites(JWTData userData, FilterSitesRequest filterSitesRequest) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    String ACTIVITY_COUNT_COLUMN = "act_count";
+
+    Condition filterCondition = noCondition();
+    if (filterSitesRequest.getTreeSpecies() != null)
+      filterCondition = filterCondition.and(SITE_ENTRIES.SPECIES.in(filterSitesRequest.getTreeSpecies()));
+    if (filterSitesRequest.getAdoptedStart() != null)
+      filterCondition = filterCondition.and(ADOPTED_SITES.DATE_ADOPTED.ge(filterSitesRequest.getAdoptedStart()));
+    if (filterSitesRequest.getAdoptedEnd() != null)
+      filterCondition = filterCondition.and(ADOPTED_SITES.DATE_ADOPTED.le(filterSitesRequest.getAdoptedEnd()));
+    if (filterSitesRequest.getNeighborhoodIds() != null)
+      filterCondition = filterCondition.and(SITES.NEIGHBORHOOD_ID.in(filterSitesRequest.getNeighborhoodIds()));
+
+    Condition stewardshipCondition = noCondition();
+    if (filterSitesRequest.getLastActivityStart() != null)
+      stewardshipCondition = stewardshipCondition.and(max(STEWARDSHIP.PERFORMED_ON).ge(filterSitesRequest.getLastActivityStart()));
+    if (filterSitesRequest.getLastActivityEnd() != null)
+      stewardshipCondition = stewardshipCondition.and(max(STEWARDSHIP.PERFORMED_ON).le(filterSitesRequest.getLastActivityEnd()));
+
+    Table<org.jooq.Record3<Integer, Integer, Integer>> activityCounts = db
```

Comment:
💭 Maybe have a comment here saying "Table containing the number of stewardship activities performed for a user on a site"?